### ISSUE TITLE
linear_regression some question

### ISSUE THREAD

**Issue by xuhang96** [aymericdamien/TensorFlow-Examples#102]
Hi,in the linear_regression 
why not
 for epoch in range(training_epochs):
        # for (x, y) in zip(train_X, train_Y):
        #     sess.run(optimizer, feed_dict={X: x, Y: y})
        sess.run(optimizer,feed_dict={X:train_X,Y:train_Y})#this is my way
I do not understand why we should use for?

**Comment by benelot** [comment]
It is a different kind of training. One is called online training where we update our weights after processing one sample. The other one is batch learning where we calculate the gradient across all samples to update the weights. Online is faster, but can end up in local minima or can take longer training time due to a possibly too coarse approximation of the real gradient of all samples. In the end it is also a bit of luck. Mini-batch learning is inbetween those two and means to learn from a limited size batch of samples to get a better approximation of the gradient. For large datasets, this can lead to a huge training speed up compared to the full-batch version.

**Comment by renziver** [comment]
I also have a question regarding Linear Regression, specifically the MSE cost function. 
why do we have to multiply the denominator(num. of instances) to 2? 

`predicted = np.array([1,2,3])`
`Y = np.array([4,5,6])`
`num_instances = predicted.shape[0]`
`cost = tf.reduce_sum(tf.pow(predicted-Y, 2))/(2*num_instances) `
`cost2 = tf.reduce_mean(tf.square(predicted - Y))`

the answer for MSE should be 9, not 4.5 but since you're multiplying the denominator by 2, the answer is becoming 4.5.
`[9+9+9]/3 = 9 `
`[9+9+9]/ 2*3 = 4.5 `
